src/services/dashboards/resource.js

```diff
@@ -330,6 +331,438 @@ const switchToTopicCentric = (input) => {
+  FROM ${tblNames.createdAroResourcesTempTableName};
+  `;
+  const pctOfReportsWithResources = sequelize.query(pctOfResourcesSql, {
+    type: QueryTypes.SELECT,
+    transaction,
+  });
+
+  // - Number of Reports with ECLKC Resources Pct -
+  const pctOfECKLKCResources = sequelize.query(`
+    WITH eclkc AS (
+      SELECT
+    COUNT(DISTINCT url) AS "eclkcCount"
+    FROM  ${tblNames.createdFlatResourceTempTableName}
+    WHERE url ilike '%eclkc.ohs.acf.hhs.gov%'
+    ), allres AS (
+    SELECT
+    COUNT(DISTINCT url) AS "allCount"
+    FROM ${tblNames.createdFlatResourceTempTableName}
+    )
+    SELECT
+      e."eclkcCount",
+    r."allCount",
+      CASE WHEN
+    r."allCount" = 0
+    THEN 0
+    ELSE round(e."eclkcCount" / r."allCount"::decimal * 100,4)
+    END AS "eclkcPct"
+    FROM eclkc e
+    JOIN allres r
+      ON 1=1;
+  `, {
+    type: QueryTypes.SELECT,
+    transaction,
+  });
+
+  // 5.) Date Headers table.
+  const dateHeaders = sequelize.query(`
+   SELECT
+    to_char("date", 'Mon-YY') AS "rollUpDate"
+   FROM ${tblNames.createdFlatResourceHeadersTempTableName};
+ `, {
+    type: QueryTypes.SELECT,
+    transaction,
+  });
+  return {
+    numberOfParticipants,
+    numberOfRecipients,
+    pctOfReportsWithResources,
+    pctOfECKLKCResources,
+    dateHeaders,
+  };
+}
+
+/*
+  Create a flat table to calculate the resource data. Use temp tables to ONLY join to the rows we need.
+  If over time the amount of data increases and slows again we can cache the flat table a set frequency.
+*/
+export async function resourceFlatData(scopes) {
+  // Date to retrieve report data from.
+  const reportCreatedAtDate = '2022-12-01';
+
+  // 1.) Get report ids using the scopes.
+  const reportIds = await ActivityReport.findAll({
+    attributes: [
+      'id',
+    ],
+    where: {
+      [Op.and]: [
+        scopes.activityReport,
+        {
+          calculatedStatus: REPORT_STATUSES.APPROVED,
+          startDate: { [Op.ne]: null },
+          createdAt: { [Op.gt]: reportCreatedAtDate },
+        },
+      ],
+    },
+    raw: true,
+  });
+
+  // Get total number of reports.
+  const totalReportCount = reportIds.length;
+
+  if (reportIds.length === 0) {
+    reportIds.push({ id: 0 });
+  }
+  // 2.) Create temp table names.
+  const createdArTempTableName = `Z_temp_resdb_ar__${uuidv4().replaceAll('-', '_')}`;
+  const createdAroResourcesTempTableName = `Z_temp_resdb_aror__${uuidv4().replaceAll('-', '_')}`;
+  const createdResourcesTempTableName = `Z_temp_resdb_res__${uuidv4().replaceAll('-', '_')}`;
+  const createdAroTopicsTempTableName = `Z_temp_resdb_arot__${uuidv4().replaceAll('-', '_')}`;
+  const createdTopicsTempTableName = `Z_temp_resdb_topics__${uuidv4().replaceAll('-', '_')}`;
+  const createdFlatResourceHeadersTempTableName = `Z_temp_resdb_headers__${uuidv4().replaceAll('-', '_')}`; // Date Headers.
+  const createdFlatResourceTempTableName = `Z_temp_resdb_flat__${uuidv4().replaceAll('-', '_')}`; // Main Flat Table.
```

Comment:
why not use the same uuid for the set of temp tables, they way it will be easier to debug when they all have the matching part of the name?